foundationdb/src/tuple.rs

```diff
@@ -404,39 +455,32 @@ impl Single for SingleValue {
             return Err(TupleError::EOF);
         }
 
-        let code = buf[0];
+        let code = SingleType::try_from(buf[0])?;
         match code {
-            0x00 => Ok((SingleValue::Empty, 1)),
-            0x01 => {
+            SingleType::Empty => Ok((SingleValue::Empty, 1)),
+            SingleType::Bytes => {
                 let (v, offset) = Single::decode(buf)?;
                 Ok((SingleValue::Bytes(v), offset))
             }
-            0x02 => {
+            SingleType::String => {
                 let (v, offset) = Single::decode(buf)?;
                 Ok((SingleValue::Str(v), offset))
             }
-            0x20 => {
+            SingleType::Float => {
                 let (v, offset) = Single::decode(buf)?;
                 Ok((SingleValue::Float(v), offset))
             }
-            0x21 => {
+            SingleType::Double => {
                 let (v, offset) = Single::decode(buf)?;
                 Ok((SingleValue::Double(v), offset))
             }
-            0x26 => Ok((SingleValue::Boolean(false), 1)),
-            0x27 => Ok((SingleValue::Boolean(false), 1)),
-            0x30 => {
+            SingleType::False => Ok((SingleValue::Boolean(false), 1)),
+            SingleType::True => Ok((SingleValue::Boolean(false), 1)),
+            SingleType::Uuid => {
                 let (v, offset) = Single::decode(buf)?;
                 Ok((SingleValue::Uuid(v), offset))
             }
-            val => {
-                if val >= 0x0b && val <= 0x1d {
-                    let (v, offset) = Single::decode(buf)?;
-                    Ok((SingleValue::Int(v), offset))
-                } else {
-                    unimplemented!("unknown code: 0x{:02x}", val);
-                }
-            }
+            _ => unimplemented!(),
```

Comment:
It will break bindingtester tests, as it requires encoding/decoding tuple integer type.

Reply:
yes, good catch, I need to go back and fix this.